### PR TITLE
added uncertainty checks to jax md

### LINKED ISSUE #319: Add MD termination criteria to Jaxmd
Now that we can compute uncertainties in Jaxmd, the next logical step would be to also include UQ termination criteria.

### PR DESCRIPTION
**Pull request by M-R-Schaefer**
Now that jaxmd supports tracking uncertainties, we can also use stopping criterions for the simulation loop.
Here I've implemented simple energy and force uncertainty stopping criterions.

closes #319

### CODE REVIEW

File: apax/md/simulate.py [hunk 1]
```diff
@@ -285,7 +297,19 @@ def body_fn(i, state):
...
+        if np.any(np.isnan(state.position)) or np.any(np.isnan(state.velocity)):
+            raise ValueError(
+                f"NaN encountered, simulation aborted after {step+1} steps."
+            )
+
+        if not all_checks_passed:
+            with logging_redirect_tqdm():
+                log.info(
+                    f"One or more dynamics checks failed at step: {step+1}"
+                )
```
Comment (PythonFZ) [inline apax/md/simulate.py:311]: A failed check is not `log.info` but more `log.warning` or even `log.critical` or `log.error`?

**Review by PythonFZ — COMMENTED** [review]
Great 🎉  one small comment. Also - any chance we can have tests for this? Probably not great without a pre-trained available ensemble model?

**Comment by M-R-Schaefer** [comment]
pre-commit.ci autofix

File: apax/md/simulate.py (hunk 1 above)
Comment (M-R-Schaefer) [reply apax/md/simulate.py:310]: Yeah fair. in IPS it is log.critical, I'll change it to that.

**Comment by M-R-Schaefer** [comment]
> Great 🎉 one small comment. Also - any chance we can have tests for this? Probably not great without a pre-trained available ensemble model?

it would be possible to test, but it takes a long time in the CI to train some model for this

**Comment by M-R-Schaefer** [comment]
pre-commit.ci autofix

**Comment by M-R-Schaefer** [comment]
yeah ok I can cook up a reasonable test for this.

**Review by PythonFZ — APPROVED** [review]

**Comment by M-R-Schaefer** [comment]
pre-commit.ci autofix

**Pull request merged**